codebundles/k8s-kubectl-namespace-healthcheck/runbook.robot

```diff
@@ -31,53 +31,84 @@ Suite Initialization
     ...    description=Which Kubernetes context to operate within.
     ...    pattern=\w*
     ...    example=my-main-cluster
+    ${RESOURCE_KINDS}=    RW.Core.Import User Variable    RESOURCE_KINDS
+    ...    type=string
+    ...    description=Which Kubernetes kinds to inspect during troubleshooting as a CSV. Depending on kinds and your cluster workloads this can increase codebundle runtime.
```

Comment:
I assume this accepts "all" as one of the options? Should we default to "all" here? or will that pull in additional resources that you don't want? e.g., if pods are returned by using "all" does that cause the taskset to evaluate pod health twice?

Reply:
`all` could be used here but we'd probably hit performance issues on larger clusters so I don't recommend it. In the code I don't have a healthcheck for pods directly as I wanted to avoid iterating through each individual pod and stay at higher level objects (eg: deployments, statefulsets)

While I do query pod logs in the events func, only pods tied to a recent event are queried, so a subset of the total pods in the namespace.

Reply:
I'm not sure why "all" would have a performance issue on larger clusters... it's scoped to the namespace, and effectively is kubectl get all (it's a pretty lightweight call)

Reply:
anyways, you've answered the question in that "all" can be used